mycroft/skills/settings.py

```diff
@@ -104,13 +106,22 @@ def __init__(self, directory, name):
         self._is_alive = True
 
         # if settingsmeta exist
-        if isfile(self._meta_path):
+        if self._meta_path:
             self._poll_skill_settings()
 
     def __hash__(self):
         """ Simple object unique hash. """
         return hash(str(id(self)) + self.name)
 
+    def _get_meta_path(self, base_directory):
```

Comment:
This could be moved outside of the class to a function

Reply:
i think this "belongs" there, makes code more readable and consistent. o you see that being used elsewhere?

i would just make it a staticmethod

Reply:
I like the static method approach as I also think this has very limited value outside this class, and think it's worth keeping it relatively close to where it's being used. I'll push up a commit with that change made. @forslund, let me know if you disagree with that approach though, and we can discuss further.

Reply:
**I'll start with the important parts:**
If a static method feels better for you guys make it so. You have (at least) as much Python experience as me so I shall not dictate code style too much.

_Then I add a rant (feel free to skip reading this):_
I personally hate static methods :) (If you want something that's a basically a function write a function)... But my roots are in C so I like functions.

I don't see it being used outside of the module but I think the module namespace is as good a container as the class for this and since it's not a part of the core functionality of the class it just feels like it's in the way. Since it's only intended to be used from within the settings.py file I feel a function is better since it won't be imported anywhere else along with the Settings class (unless specifically.

Note: Above statements are exaggerated for comic effect but is basically how I feel. I do think static methods can be useful.

Reply:
With only one class in the file, top level or static is more or less the same, I suppose. I don't feel particularly strongly, so I've moved it to the bottom of the file, as a module level function. @forslund, let me know if there's anything else you'd like me to tweak.